settings/helpers.py

```diff
@@ -105,7 +105,7 @@ def send_email(data: SingleEmailInput) -> SingleEmailSuccessResponse | SingleEma
 
         try:
             if isinstance(data.content, dict):
-                data_str: str = (
+                data_str: str | Any = (
```

Comment:
This should always be a string though

Reply:
True but he sees that in `SingleTemplatedEmailContent` `template_data` is set to `dict | str` and is trowing an error suggesting a type of `dict[Any, Any] | str | None` and `reveal_type` is suggesting the same. So if we can't change or extend type of any of them solution that I see at the moment is to use `# type: ignore[assignment]` OR leave it without type annotation since it's named `data_str` and is used only once.

Reply:
I think it's due to `SingleTemplatedEmailContent` and that `template_data` has also `dict` type, because when i left it with only `str` type error was gone from `data_str` but it moved to next line 109 and complain that `data.content.get("template_data")` return type is `str | None` instead of `str`.

So to keep type `str` we need to put `# type: ignore[assignment]` on line 108 and 109 because 109 got return type of `dict[Any, Any] | str | None` in this case.

Reply:
Ah so I guess instead of data.content.get("template_data") we was meant to use data_str? I just didn't use the variable?

Reply:
No `data_str` is used once later on in content of `send_email` and we need `data.content.get("template_data")` to assign it's value to `data_str` when it's just a string but `mypy` assume that `data.content.get("template_data")` might be empty and can return only `None` so he complains that `data_str` demands type `str` but `.get` return type is `str | None`.

I would suggest to leave `data_str` with out type since `SingleTemplatedEmailContent` is already covering `template_data` as `str`, we might only add in `SingleTemplatedEmailContent` to template_data `dict[str, str] | str` instead of just `dict | str` so it will cover that the content of dict should also be a string.

Reply:
Let me know which option would you prefer and I will push the changes.

Reply:
> I would suggest to leave data_str with out type since SingleTemplatedEmailContent is already covering template_data as str, we might only add in SingleTemplatedEmailContent to template_data dict[str, str] | str instead of just dict | str so it will cover that the content of dict should also be a string.

Yeah that sounds good to me, apologies for the late reply.

It's odd cause
```py
data_str: str | Any = (
                    data.content.get("template_data")
                    if isinstance(data.content.get("template_data"), str)
                    else json.dumps(data.content.get("template_data"))
)
```
will always be a string no matter what, bit odd that mypy flags that. Thanks for the resolution though

Reply:
No problem ;) When I'll be on my working station I'll make this change, push it and will let you know when it's ready.